src/quest/persistence.py

```diff
@@ -79,6 +91,171 @@ def has_blob(self, key: str) -> bool:
     def delete_blob(self, key: str):
         self._get_file(key).unlink()
 
+Base = declarative_base()
+
+class RecordModel(Base):
+    __tablename__ = 'records'
+
+    id = Column(Integer, primary_key=True, autoincrement=True)
+    name = Column(String) # TODO good name for this?
+    key = Column(String)
+    blob = Column(JSON)
+
+    def __repr__(self):
+        return f'<{self.__class__.__name__}: {self.name}>'
+
+class SQLDatabase:
+
+    def __init__(self, db_url: str):
+        self._db_url = db_url
+        self._engine = create_engine(db_url)
+
+        Base.metadata.create_all(self._engine)
+
+    def get_engine(self) -> Engine:
+        return self._engine
+
+class SqlBlobStorage(BlobStorage):
+    def __init__(self, name, engine):
+        self._name = name
+        self._engine = engine
+
+    def _get_session(self):
+        return sessionmaker(bind=self._engine)()
+
+    def write_blob(self, key: str, blob: Blob):
+        # Check to see if a blob exists, if so rewrite it
+        with self._get_session() as session:
```

Comment:
Do we want to be opening so many sessions? I would assume that typically only one session per runtime makes sense. If the `SqlBlobStorage` needs to support the context API to handle this correctly, we can do that.

Reply:
Done